rfd/0169-auto-updates-linux-agents.md

```diff
@@ -91,45 +89,145 @@ spec:
   # cluster level. Only turn agent automatic updates off if self-managed
   # agent updates are in place.
   agent_auto_update: true|false
-  # agent_update_hour sets the hour in UTC at which clients should update their agents.
-  agent_update_hour: 0-23
-  # agent_update_jitter_seconds sets a duration in which the upgrade will occur after the hour.
-  # The agent upgrader will pick a random time within this duration to wait to upgrade.
-  agent_update_jitter_seconds: 0-3600
-  
-  [...]
+
+  # agent_auto_update_groups contains both "regular" and "critical" schedules.
+  # The schedule used is determined by the agent_version_schedule associated
+  # with the version in autoupdate_version.
+  # Groups are not configurable with the "immediate" schedule.
+  agent_auto_update_groups:
+    # schedule is "regular" or "critical"
+    regular:
+    - name: staging-group
+      # agent_selection defines which agents are included in the group.
+      agent_selection:
+        # query selects agents by resource query.
+        #  default: all connected agents
+        query: 'labels["environment"]=="staging"'
```

Comment:
Labels are per-service, what does it mean to query agents by label?

Reply:
Mentioned here: https://github.com/gravitational/teleport/pull/40190/files#diff-4fb277bd3d884c7b045e7efad61111ac4e7957c063e0c0c9281f4e74fd72997fR58

If the label is present on any connected resources provided by the agent, the agent is eligible.

The intent of the UX is to allow customers to think about upgrades in terms of connected resources that might be affected by the upgrade.